### PR TITLE
Convert genFile() to use a high-resolution, 1-period lookup table

### PR DESCRIPTION
**Pull request by freitagbr**
### File Generation

Implemented a high-resolution 1-period lookup table, which allows:
- Reduced memory usage
- Higher fidelity
  - High resolution lookup table reduces distortion
  - Default size of lookup table is 8192 samples
  - For best sound, do not set the size any lower than 2048
##### Note

I tried a 1/4 period lookup table, but in practice the performance was worse.
### Housekeeping
- Removed unnecessary comments. Leave comments for whole blocks or tricky lines of code.
- Renamed some variables.
- Implemented some style guidelines, according to the cpplint linter; ignored the line length policy.

### CODE REVIEW

File: tuningFork.c [hunk 1]
```diff
@@ -203,93 +186,61 @@ void genFile(double freq, double duration, char* fname)
...
-    {
-        //fprintf(stderr, "omegaT is %lf\n", omegaT);
-        sample = max_volume * sin(omegaT);
-        //sample = max_volume * sin(omega * (double)i/SAMPLE_RATE);
-
-        sampleTable[i] = sample;
-
-        omegaT += deltaOmegaT; // gives next value to evaluate sin at
+    const int lookupTableSize = 8192;  // Size in samples
+    const int quarterTableSize = lookupTableSize >> 2;
```
Comment (nfischer) [inline tuningFork.c]: Change ">> 2" to "/ 4" please. The meaning will be more clear, and our compiler will convert it to the more efficient operation

File: tuningFork.c [hunk 2]
```diff
@@ -203,93 +186,61 @@ void genFile(double freq, double duration, char* fname)
...
-    for (i = 0; i < tableSize; i++)
-    {
-        //fprintf(stderr, "omegaT is %lf\n", omegaT);
-        sample = max_volume * sin(omegaT);
-        //sample = max_volume * sin(omega * (double)i/SAMPLE_RATE);
-
-        sampleTable[i] = sample;
-
-        omegaT += deltaOmegaT; // gives next value to evaluate sin at
+    const int lookupTableSize = 8192;  // Size in samples
```
Comment (nfischer) [inline tuningFork.c]: Where does 8192 come from?

File: tuningFork.c [hunk 3]
```diff
@@ -203,93 +186,61 @@ void genFile(double freq, double duration, char* fname)
...
-            periodIndex = 0;
-        }
+    const double phaseIncrement = (double)(freq / (2.0 * SAMPLE_RATE)) * lookupTableSize;
+    double phase = 0.0;
+    int phase_i = 0;
+    int16_t* dataSubChunk = malloc(sizeof(int16_t) * numSamples * 2);
+
+    for (int i = 0; i < numSamples; ++i) {
+        phase_i = (int)phase;
+        int j = i * 2;
```
Comment (nfischer) [inline tuningFork.c]: Could we move j out of the loop and increment it by 2 at each iteration instead? That would run more efficiently I think

File: tuningFork.c [hunk 4]
```diff
@@ -203,93 +186,61 @@ void genFile(double freq, double duration, char* fname)
...
-        sampleTable[i] = sample;
-
-        omegaT += deltaOmegaT; // gives next value to evaluate sin at
+    const int lookupTableSize = 8192;  // Size in samples
+    const int quarterTableSize = lookupTableSize >> 2;
+    int16_t* lookupTable = malloc(sizeof(int16_t) * lookupTableSize);
+
+    // Generate lookup table for 1/4 period of a sine wave
+    // We can evaluate any value over an entire period using just the first quadrant
+    for (int i = 0; i < quarterTableSize; i++) {
```
Comment (nfischer) [inline tuningFork.c]: Change this to:

```
int i;
for (i=0; i < quarterTableSize; i++) {
```

This is to stay consistent with older C standards (and avoid compiling with the `-std=C99` flag)

**Comment by nfischer** [comment]
I left some comments in your file for some things that should be fixed before this is ready to merge.

First off, this is not suitable to merge because it doesn't compile on Linux systems (please see my comments about C99).

Next, this doesn't seem to generate the correct tone. It sounds too low to me. Please compare it against the master branch.

On the bright side, it seems to generate the file much faster, which is good.

File: tuningFork.c (hunk 2 above)
Comment (freitagbr) [inline tuningFork.c]: 8192 is an arbitrary size that I chose. A full period of 8192 has a frequency of about 5Hz at a sample rate of 44100, and 5Hz is much lower than the range of human hearing. This should reduce the effect of aliasing. You can set it higher or lower if you want, though I would not set it lower than 2048, or about 20Hz (within human hearing range).

**Comment by nfischer** [comment]
Can't listen to it right now, but it looks like the frequency spectrums don't match. Could you take a look into this? Not sure which is a cleaner sound.

**Comment by nfischer** [comment]
I think this needs to wait until #3 is resolved (I have pending PR #4 to resolve this).

**Comment by nfischer** [comment]
@freitagbr Try merging/rebasing this off `master`. Then we can try out the testing framework.

**Comment by nfischer** [comment]
So I just checked it. It now passes all tests. Pulling the file up in Audacity doesn't make it look any more accurate than the master branch. And the benchmarks seem roughly the same.

The memory savings don't seem significant, since the bulk of memory is in the file buffer, not the lookup table. Correct me if I'm wrong.

**Comment by nfischer** [comment]
@freitagbr Do you think this has significant improvements over the previous version?

**Comment by freitagbr** [comment]
I think there are performance improvements, but they're insignificant given our powerful hardware. It might be more worth working on a sine-generator that is mathematically correct instead of lookup-table based.

**Comment by nfischer** [comment]
One idea that I had would be a lookup table that has a finer resolution (up to 4x). Might not be faster, but it might give more accurate results. Something to consider if we're trying to reduce distortion and still have fairly good performance.

**Comment by nfischer** [comment]
@freitagbr is this worth pursuing?

**Pull request closed without merging**